pkg/storage/allocator.go

```diff
@@ -513,7 +515,7 @@ func (a *Allocator) ShouldTransferLease(
 		log.Infof(ctx, "ShouldTransferLease (lease-holder=%d):\n%s", leaseStoreID, sl)
 	}
 
-	transferDec, _ := a.shouldTransferLeaseUsingStats(ctx, sl, source, existing, stats)
+	transferDec, _ := a.shouldTransferLeaseUsingStats(ctx, sl, source, existing, stats, wantScatter)
```

Comment:
Even if we do want to use `wantScatter` within `ShouldTransferLease`, we don't really need to pass `wantScatter` down - just use it to override the `shouldNotTransfer` result.